staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -182,6 +182,10 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 		return &roundTripper{a, rt}
 	})
 
+	if c.HasTokenAuth() {
+		return nil
+	}
```

Comment:
Thinking on this a bit more, maybe this should be the first line in this function.  As it stands, it attempts to configure token auth via the `c.Wrap` call but then skips the cert auth.

It may make more sense for `UpdateTransportConfig` to do nothing when `c.HasTokenAuth()` returns `true` instead of doing "half" of what it is supposed to do.  We still want the `req.Header.Get("Authorization") != ""` logic below because round trippers can be arbitrarily stacked via `c.Wrap`.

Reply:
Agreed - moved to top of function.

Reply:
Agreed - done.